cmd/zc_traverser_blob.go

```diff
@@ -95,6 +98,26 @@ func (t *blobTraverser) getPropertiesIfSingleBlob() (props *azblob.BlobGetProper
 	return nil, false, false, err
 }
 
+func (t *blobTraverser) getBlobTags(requestID *string, snapshot *string, versionID *string, ifTags *string) (common.BlobTags, error) {
+	// trim away the trailing slash before we check whether it's a single blob
+	// so that we can detect the directory stub in case there is one
+	blobUrlParts := azblob.NewBlobURLParts(*t.rawURL)
+	blobUrlParts.BlobName = strings.TrimSuffix(blobUrlParts.BlobName, common.AZCOPY_PATH_SEPARATOR_STRING)
+
+	// perform the check
+	blobURL := azblob.NewBlobURL(blobUrlParts.URL(), t.p)
+	blobTagsMap := make(common.BlobTags)
+	blobGetTagsResp, err := blobURL.GetTags(t.ctx, nil, requestID, snapshot, versionID, ifTags)
```

Comment:
>snapshot, versionID [](start = 64, length = 19)

shouldn't snapshot and version ID be part of the blob URL already? Why do we need to pass them separately?